### ISSUE TITLE
Confusing usage of term `local` in DG kernels

### ISSUE THREAD

**Issue by jkozdon** [CliMA/ClimateMachine.jl#1565]
I've noticed in the kernels we use the prefix `local` to mean both uniform memory and private memory.

We also use the term `shared` memory for what KA calls `local` memory.

I imagine that this could make navigating / reading the kernels harder for someone new to them (you know because it's so easy right now anyway 🤣)

Not sure if we want some form of consistent naming climate machine wide for kernel memory types.

Thoughts?

The easiest thing would be to prefix with the KA naming:
- `@uniform` -> `u_` or `uniform_`
- `@private` -> `p_` or `private_`
- `@localmem` -> `l_` or `local_` (or maybe `localmn_` so it is 7 characters like the other two?)

**Comment by blallen** [comment]
Yes please do something like this. I can follow the DG-ness of the kernels but not the GPU memory management and this would help so much.

Also in favor of the longer prefix to help new people

**Comment by jkozdon** [comment]
> Also in favor of the longer prefix to help new people

Personally I prefer shorter prefixes, since the kernels already look nasty with the formatter, longer names would make this worse.

**Comment by jkozdon** [comment]
> > Also in favor of the longer prefix to help new people
> 
> Personally I prefer shorter prefixes, since the kernels already look nasty with the formatter, longer names would make this worse.

I think that it's also OK to use both, it's not really that confusing in this context. I think the bigger point of confusion is that the term `local` used to mean three different things!

Though I do not think that we ever refer to KA `localmem` as local. This is almost exclusively called `shared` memory in the codes prefixing system which is hold over GPUifyLoops and CUDA terminology.

**Comment by charleskawczynski** [comment]
Another thing we could do is use long names at the start of the kernels and assign short name aliases. I don't mind this pattern because users can still trace things to see the long names.

**Comment by blallen** [comment]
> Another thing we could do is use long names at the start of the kernels and assign short name aliases. I don't mind this pattern because users can still trace things to see the long names.

Good idea. I now just default to long names because I assume every function is going to be one argument per line anyways and at that point, everything can have a 50 character name
